pkg/controller/job/job_controller_test.go

```diff
@@ -1249,6 +1250,10 @@ func TestControllerSyncJob(t *testing.T) {
 	for name, tc := range testCases {
 		t.Run(name, func(t *testing.T) {
 			logger, _ := ktesting.NewTestContext(t)
+			if tc.podIndexLabelDisabled {
```

Comment:
I'm missing similar changes in the statefulset controller, looking [here](https://github.com/kubernetes/kubernetes/blob/b8e20b74ddc9d0f072436e569877429ea6efa799/pkg/controller/statefulset/stateful_set_control_test.go#L233-L256) we're only verifying the labels when the feature gate is on. We should rather check for existence when the feature is on, and check for them not being there when it's off.

Reply:
cc @alaypatel07 

Let's try and address this so we can get this in without any extensions.